web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -167,55 +173,220 @@ file systems will be created on the selected device.");
 };
 
 /**
- * Allows to select LVM
+ * Form for configuring the system volume group.
+ * @component
+ *
+ * @param {object} props
+ * @param {string} props.id - Form ID.
+ * @param {ProposalSettings} props.settings - Settings used for calculating a proposal.
+ * @param {StorageDevice[]} [props.devices=[]] - Available storage devices.
+ * @param {onSubmitFn} [props.onSubmit=noop] - On submit callback.
+ * @param {onValidateFn} [props.onValidate=noop] - On validate callback.
+ *
+ * @callback onSubmitFn
+ * @param {string[]} devices - Name of the selected devices.
+ *
+ * @callback onValidateFn
+ * @param {boolean} valid
+ */
+const LVMSettingsForm = ({
+  id,
+  settings,
+  devices = [],
+  onSubmit: onSubmitProp = noop,
+  onValidate = noop
+}) => {
+  const [vgDevices, setVgDevices] = useState(settings.systemVGDevices);
+  const [isBootDeviceSelected, setIsBootDeviceSelected] = useState(settings.systemVGDevices.length === 0);
+  const [editedDevices, setEditedDevices] = useState(false);
+
+  const selectBootDevice = () => {
+    setIsBootDeviceSelected(true);
+    onValidate(true);
+  };
+
+  const selectCustomDevices = () => {
+    setIsBootDeviceSelected(false);
+    const { bootDevice } = settings;
+    const customDevices = (vgDevices.length === 0 && !editedDevices) ? [bootDevice] : vgDevices;
+    setVgDevices(customDevices);
+    onValidate(customDevices.length > 0);
+  };
+
+  const onChangeDevices = (devices) => {
+    setVgDevices(devices);
+    setEditedDevices(true);
+    onValidate(devices.length > 0);
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    const customDevices = isBootDeviceSelected ? [] : vgDevices;
+    onSubmitProp(customDevices);
+  };
+
+  const BootDevice = () => {
+    const bootDevice = devices.find(d => d.name === settings.bootDevice);
+
+    return <DeviceList devices={[bootDevice]} />;
+  };
+
+  return (
+    <Form id={id} onSubmit={onSubmit}>
+      <div className="split">
+        <span>{_("Devices for creating the volume group")}</span>
+        <ToggleGroup aria-label="Volume group devices" isCompact>
```

Comment:
NP: I'm not sure about the use of the aria-label here. It's not required for that component and we already have an text before it that I guess is enough